boost-1.74 has a buggy serialization/set.hpp header.
This was fixed in 1.75.

--- src/serialization.cc~	2021-04-21 09:22:18.000000000 +0200
+++ src/serialization.cc	2021-05-23 15:00:06.216308457 +0200
@@ -15,6 +15,10 @@
 // received a copy of the GNU Lesser General Public License along with
 // hpp-manipulation  If not, see
 // <http://www.gnu.org/licenses/>.
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 107400 && BOOST_VERSION < 107500
+# include <boost/serialization/library_version_type.hpp>
+#endif
 
 #include <boost/serialization/serialization.hpp>
 #include <boost/serialization/set.hpp>
